tests/e2e/test_e2e.py

```diff
@@ -80,6 +80,21 @@ def test_e2e_default_batching():
     assert resp.json() == {"output": 16.0}, "tests/simple_server.py didn't return expected output"
 
 
+@e2e_from_file("tests/e2e/default_exc_handler.py")
+def test_e2e_default_exc_handler():
+    resp = requests.post("http://127.0.0.1:8000/predict", json={"base64": "_9j_4AAQSkZJRgABAQEAYAB"}, headers=None)
+    assert resp.status_code == 400, f"Expected response to be 400 but got {resp.status_code}"
+    assert resp.json() == {"message": "cannot decode input"}, "server didn't return expected output"
+
+
+@e2e_from_file("tests/e2e/default_middleware.py")
```

Comment:
Please create a dummy middleware, pass that to LitServer and assert the test.

Reply:
Do you mean I should launch a LitServer with some dummy middleware that does nothing and add a test that checks if it responds without errors?

Reply:
yes, maybe the dummy middleware can modify the request data and you can assert it in LitAPI.decode_method. 

Example:

* the client input is `{"input": 4.0}`
* Middleware changes it to `{"input": 5.0}`
* LitAPI.decode_method `assert request["input"]==5.0`

Reply:
I think we can move the class definition to `tests/test_lit_server.py` and then delete `tests/e2e/default_middleware.py`. You can write a test similar to [this](https://github.com/Lightning-AI/LitServe/blob/74d80e78af12411e05054da7a1b62fefb8135106/tests/test_lit_server.py#L454).

Reply:
ok, I moved the test from e2e to test_lit_server.